src/bin/gpfdist/transform.c

```diff
@@ -781,4 +781,31 @@ transform_saferegex(struct transform* tr)
 	return &(tr->saferegex);
 }
 
+/*
+ * validate transformation name from options
+ */
+
+const char*
+validate_transform_opt(struct transform* trlist, const char* name)
+{
+	size_t name_len;
+
+	if (!trlist)
+		return "Error: default transformation is set,"
+			" but TRANSFORMATIONS are not specified";
+
+	name_len = strlen(name);
+
+	/*
+	 * Transformation names are parsed from config file and stored as map
+	 * keys. Validate the same key len here. Correspondence of default
+	 * transformation to all transformations is checked at runtime.
+	 */
+	if (name_len < 1 || name_len > MAX_KEYLEN)
+		return "Error: -X default transformation name length must be"
```

Comment:
I guess the `-X` here is a rudiment of the previous patch version and the error message should also be updated to use `-I` and `-O` instead of `-X`.

Getting `-X` in error message might confuse user, as there is no such option now.

Reply:
Fixed